python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -1897,3 +1895,64 @@ def unique_id_from_asset_and_check_keys(entity_keys: Iterable["EntityKey"]) -> s
     """
     sorted_key_strs = sorted(str(key) for key in entity_keys)
     return non_secure_md5_hash_str(json.dumps(sorted_key_strs).encode("utf-8"))[:8]
+
+
+def replace_specs_on_asset(
+    assets_def: AssetsDefinition, replaced_specs: Sequence[AssetSpec]
+) -> "AssetsDefinition":
+    from dagster._builtins import Nothing
+    from dagster._core.definitions.input import In
+
+    new_deps = set().union(*(spec.deps for spec in replaced_specs))
+    previous_deps = set().union(*(spec.deps for spec in assets_def.specs))
+    added_deps = new_deps - previous_deps
+    removed_deps = previous_deps - new_deps
+    remaining_original_deps = previous_deps - removed_deps
+    original_key_to_input_mapping = reverse_dict(assets_def.node_keys_by_input_name)
+
+    # If there are no changes to the dependency structure, return the original assets_def
```

Comment:
nit: slightly confusing comment as we're not returning the original assets_def (we're returning an assets def with the original node)

Reply:
oof you right

Reply:
fixed